apps/account-api/src/services/siwfV2.service.spec.ts

```diff
@@ -0,0 +1,137 @@
+    process.env.GRAPH_ENVIRONMENT_TYPE = 'TestnetPaseo';
+  });
+
+  beforeEach(async () => {
+    const module: TestingModule = await Test.createTestingModule({
+      imports: [
+        ConfigModule.forRoot({
+          load: [apiConfig, blockchainConfig],
+        }),
+      ],
+      providers: [SiwfV2Service, mockBlockchainServiceProvider],
+    }).compile();
+
+    siwfV2Service = module.get<SiwfV2Service>(SiwfV2Service);
+  });
+
+  it('should be defined', () => {
+    expect(siwfV2Service).toBeDefined();
+  });
+
+  describe('getRedirectUrl', () => {
+    let result: WalletV2RedirectResponseDto;
+
+    beforeAll(async () => {
+      result = await siwfV2Service.getRedirectUrl(exampleCallback, examplePermissions, exampleCredentials);
+    });
+
+    it('Should return a valid signedRequest', () => {
+      // Assert
+      expect(result).toHaveProperty('signedRequest');
+
+      const parsedResult = decodeSignedRequest(result.signedRequest);
+      expect(parsedResult.requestedSignatures.payload.callback).toEqual(exampleCallback);
+    });
+
+    it('Should return the signedRequest with credentials', () => {
+      // Assert
+      expect(result).toHaveProperty('signedRequest');
+
+      const parsedResult = decodeSignedRequest(result.signedRequest);
+      expect(parsedResult.requestedCredentials).toHaveLength(2);
+    });
+
+    it('Should return the signedRequest with permissions', () => {
+      // Assert
+      expect(result).toHaveProperty('signedRequest');
+
+      const parsedResult = decodeSignedRequest(result.signedRequest);
+      expect(parsedResult.requestedSignatures.payload.permissions).toEqual(examplePermissions);
+    });
+
+    it('Should go to the correct redirect url', () => {
+      const resultUrl = new URL(result.redirectUrl);
+
+      expect(resultUrl.pathname).toEqual('/siwf/start');
+      expect(resultUrl.host).toEqual('www.example.com');
+    });
+  });
+
+  describe('swifV2Endpoint', () => {
+    let siwfV2ServiceEndpointTest: SiwfV2Service;
+    beforeAll(() => {
+      process.env.FREQUENCY_URL = 'http://127.0.0.1:9944';
+      process.env.FREQUENCY_HTTP_URL = 'http://127.0.0.1:9944';
+      process.env.PROVIDER_ACCOUNT_SEED_PHRASE =
+        'offer debate skin describe light badge fish turtle actual inject struggle border';
+      process.env.PROVIDER_ID = '1';
+      process.env.WEBHOOK_BASE_URL = 'http://127.0.0.1';
+      process.env.CAPACITY_LIMIT = '{"type":"amount","value":"80"}';
+      process.env.GRAPH_ENVIRONMENT_TYPE = 'TestnetPaseo';
+    });
+
+    beforeEach(async () => {
+      process.env.SIWF_V2_URL = '';
+      const module: TestingModule = await Test.createTestingModule({
+        imports: [
+          ConfigModule.forRoot({
+            load: [apiConfig, blockchainConfig],
+          }),
+        ],
+        providers: [SiwfV2Service, mockBlockchainServiceProvider],
+      }).compile();
+
+      siwfV2ServiceEndpointTest = module.get<SiwfV2Service>(SiwfV2Service);
+    });
```

Comment:
All this duplicate test harness setup is overkill, just to control the config environment. Take a look at `testlib/utils.config-tests.ts`. `GenerateMockConfigProvider` can be used to mock the config & enable using `jest.spyon(mockConfigProvider, 'configValue', 'get').mockReturnValue(...)` in specific tests.